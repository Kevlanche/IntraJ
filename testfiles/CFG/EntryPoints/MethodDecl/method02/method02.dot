digraph G {
graph [splines=ortho, nodesep="1", ranksep="1"]
node [shape=rect, fontname=Arial];
{ rank = same; "Program[0]"}
{ rank = same; "List[1]"}
{ rank = same; "CompilationUnit[2]"}
{ rank = same; "List[3]"; "List[4]"}
{ rank = same; "ClassDecl[5]"}
{ rank = same; "Modifiers[6]"; "Opt[9]"; "List[10]"; "List[11]"}
{ rank = same; "List[7]"; "MethodDecl[12]"}
{ rank = same; "Modifier[8]"; "Entry[13]"; "Modifiers[14]"; "PrimitiveTypeAccess[16]"; "List[17]"; "List[26]"; "Opt[27]"; "Exit[28]"}
{ rank = same; "List[15]"; "ParameterDeclaration[18]"; "ParameterDeclaration[22]"; "Block[28]"}
{ rank = same; "Modifiers[19]"; "PrimitiveTypeAccess[21]"; "Modifiers[23]"; "PrimitiveTypeAccess[25]"; "List[29]"}
{ rank = same; "List[20]"; "List[24]"; "ExprStmt[30]"}
{ rank = same; "Dot[31]"}
{ rank = same; "Dot[32]"; "MethodAccess[35]"}
{ rank = same; "TypeAccess[33]"; "VarAccess[34]"; "List[37]"; "List[41]"; "List[42]"; "UncheckedExceptions[36]"}
{ rank = same; "AddExpr[38]"}
{ rank = same; "VarAccess[39]"; "VarAccess[40]"}
"Program[0]" [label="Program\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[1]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"CompilationUnit[2]" [label="CompilationUnit\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[3]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[4]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ClassDecl[5]" [label="ClassDecl\n < method02 >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[6]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[7]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifier[8]" [label="Modifier\n < public >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[9]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[10]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[11]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"MethodDecl[12]" [label="MethodDecl\n < m >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Entry[13]" [label="Entry\n ", fillcolor=white   style=filled  ]
"Modifiers[14]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[15]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[16]" [label="PrimitiveTypeAccess\n < void >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[17]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ParameterDeclaration[18]" [label="ParameterDeclaration\n < e >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[19]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[20]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[21]" [label="PrimitiveTypeAccess\n < int >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ParameterDeclaration[22]" [label="ParameterDeclaration\n < g >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[23]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[24]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[25]" [label="PrimitiveTypeAccess\n < int >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[26]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[27]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Block[28]" [label="Block\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[29]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ExprStmt[30]" [label="ExprStmt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Dot[31]" [label="Dot\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Dot[32]" [label="Dot\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"TypeAccess[33]" [label="TypeAccess\n < System >", fillcolor=white   style=filled  ]
"VarAccess[34]" [label="VarAccess\n < out >", fillcolor=white   style=filled  ]
"MethodAccess[35]" [label="MethodAccess\n < println >", fillcolor=white   style=filled  ]
"List[37]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"AddExpr[38]" [label="AddExpr\n < + >", fillcolor=white   style=filled  ]
"VarAccess[39]" [label="VarAccess\n < e >", fillcolor=white   style=filled  ]
"VarAccess[40]" [label="VarAccess\n < g >", fillcolor=white   style=filled  ]
"List[41]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[42]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"UncheckedExceptions[36]" [label="UncheckedExceptions\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Exit[28]" [label="Exit\n ", fillcolor=white   style=filled  ]
"CompilationUnit[2]" -> "List[3]" [style=dashed, color=gray]
"List[7]" -> "Modifier[8]" [style=dashed, color=gray]
"Modifiers[6]" -> "List[7]" [style=dashed, color=gray]
"ClassDecl[5]" -> "Modifiers[6]" [style=dashed, color=gray]
"ClassDecl[5]" -> "Opt[9]" [style=dashed, color=gray]
"ClassDecl[5]" -> "List[10]" [style=dashed, color=gray]
"MethodDecl[12]" -> "Entry[13]" [style=dashed, color=gray]
"Modifiers[14]" -> "List[15]" [style=dashed, color=gray]
"MethodDecl[12]" -> "Modifiers[14]" [style=dashed, color=gray]
"MethodDecl[12]" -> "PrimitiveTypeAccess[16]" [style=dashed, color=gray]
"Modifiers[19]" -> "List[20]" [style=dashed, color=gray]
"ParameterDeclaration[18]" -> "Modifiers[19]" [style=dashed, color=gray]
"ParameterDeclaration[18]" -> "PrimitiveTypeAccess[21]" [style=dashed, color=gray]
"List[17]" -> "ParameterDeclaration[18]" [style=dashed, color=gray]
"Modifiers[23]" -> "List[24]" [style=dashed, color=gray]
"ParameterDeclaration[22]" -> "Modifiers[23]" [style=dashed, color=gray]
"ParameterDeclaration[22]" -> "PrimitiveTypeAccess[25]" [style=dashed, color=gray]
"List[17]" -> "ParameterDeclaration[22]" [style=dashed, color=gray]
"MethodDecl[12]" -> "List[17]" [style=dashed, color=gray]
"MethodDecl[12]" -> "List[26]" [style=dashed, color=gray]
"Dot[32]" -> "TypeAccess[33]" [style=dashed, color=gray]
"Dot[32]" -> "VarAccess[34]" [style=dashed, color=gray]
"Dot[31]" -> "Dot[32]" [style=dashed, color=gray]
"AddExpr[38]" -> "VarAccess[39]" [style=dashed, color=gray]
"AddExpr[38]" -> "VarAccess[40]" [style=dashed, color=gray]
"List[37]" -> "AddExpr[38]" [style=dashed, color=gray]
"MethodAccess[35]" -> "List[37]" [style=dashed, color=gray]
"MethodAccess[35]" -> "List[41]" [style=dashed, color=gray]
"MethodAccess[35]" -> "List[42]" [style=dashed, color=gray]
"MethodAccess[35]" -> "UncheckedExceptions[36]" [style=dashed, color=gray]
"Dot[31]" -> "MethodAccess[35]" [style=dashed, color=gray]
"ExprStmt[30]" -> "Dot[31]" [style=dashed, color=gray]
"List[29]" -> "ExprStmt[30]" [style=dashed, color=gray]
"Block[28]" -> "List[29]" [style=dashed, color=gray]
"Opt[27]" -> "Block[28]" [style=dashed, color=gray]
"MethodDecl[12]" -> "Opt[27]" [style=dashed, color=gray]
"MethodDecl[12]" -> "Exit[28]" [style=dashed, color=gray]
"List[11]" -> "MethodDecl[12]" [style=dashed, color=gray]
"ClassDecl[5]" -> "List[11]" [style=dashed, color=gray]
"List[4]" -> "ClassDecl[5]" [style=dashed, color=gray]
"CompilationUnit[2]" -> "List[4]" [style=dashed, color=gray]
"List[1]" -> "CompilationUnit[2]" [style=dashed, color=gray]
"Program[0]" -> "List[1]" [style=dashed, color=gray]
"Exit[28]" -> "MethodAccess[35]" [color=red, constraint=false]
"MethodAccess[35]" -> "Exit[28]" [color=blue, constraint=false, xlabel=" "] 
"MethodAccess[35]" -> "AddExpr[38]" [color=red, constraint=false]
"AddExpr[38]" -> "MethodAccess[35]" [color=blue, constraint=false, xlabel=" "] 
"AddExpr[38]" -> "VarAccess[40]" [color=red, constraint=false]
"VarAccess[40]" -> "AddExpr[38]" [color=blue, constraint=false, xlabel=" "] 
"VarAccess[40]" -> "VarAccess[39]" [color=red, constraint=false]
"VarAccess[39]" -> "VarAccess[40]" [color=blue, constraint=false, xlabel=" "] 
"VarAccess[39]" -> "VarAccess[34]" [color=red, constraint=false]
"VarAccess[34]" -> "VarAccess[39]" [color=blue, constraint=false, xlabel=" "] 
"VarAccess[34]" -> "TypeAccess[33]" [color=red, constraint=false]
"TypeAccess[33]" -> "VarAccess[34]" [color=blue, constraint=false, xlabel=" "] 
"TypeAccess[33]" -> "Entry[13]" [color=red, constraint=false]
"Entry[13]" -> "TypeAccess[33]" [color=blue, constraint=false, xlabel=" "] 
"List[3]" -> "List[4]" [style=invis]
"Modifiers[6]" -> "Opt[9]" -> "List[10]" -> "List[11]" [style=invis]
"Entry[13]" -> "Modifiers[14]" -> "PrimitiveTypeAccess[16]" -> "List[17]" -> "List[26]" -> "Opt[27]" -> "Exit[28]" [style=invis]
"ParameterDeclaration[18]" -> "ParameterDeclaration[22]" [style=invis]
"Modifiers[19]" -> "PrimitiveTypeAccess[21]" [style=invis]
"Modifiers[23]" -> "PrimitiveTypeAccess[25]" [style=invis]
"Dot[32]" -> "MethodAccess[35]" [style=invis]
"TypeAccess[33]" -> "VarAccess[34]" [style=invis]
"VarAccess[39]" -> "VarAccess[40]" [style=invis]

subgraph cluster_legend {
{node [style=filled, fillcolor=1, colorscheme="pastel13"]
Nodes [label="Total Nr. Nodes
44", fillcolor=1]
Edges [label="Total Nr. Edges
57", fillcolor=1]
NodesCFG [label="CFG Nr. Nodes
8", fillcolor=3]
EdgesCFG [label="CFG Nr. Edges
7", fillcolor=3]
Nodes-> "Program[0]" -> NodesCFG  [style="invis"]
Edges -> "Program[0]" ->EdgesCFG [style="invis"]}
label = "Statistics of the framework: NEXTFramework"
style="solid"
ranksep=0.05
nodesep=0.01
labelloc = b
len=2
}}