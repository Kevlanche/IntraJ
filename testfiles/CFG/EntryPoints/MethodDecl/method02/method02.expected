NodeName: Entry[13] Successors: [TypeAccess[33]]
NodeName: TypeAccess[33] Successors: [VarAccess[34]]
NodeName: VarAccess[34] Successors: [VarAccess[39]]
NodeName: VarAccess[39] Successors: [VarAccess[40]]
NodeName: VarAccess[40] Successors: [AddExpr[38]]
NodeName: AddExpr[38] Successors: [MethodAccess[35]]
NodeName: MethodAccess[35] Successors: [Exit[28]]
NodeName: Exit[28] Successors: []

NodeName: Exit[28] Predecessor: [MethodAccess[35]]
NodeName: MethodAccess[35] Predecessor: [AddExpr[38]]
NodeName: AddExpr[38] Predecessor: [VarAccess[40]]
NodeName: VarAccess[40] Predecessor: [VarAccess[39]]
NodeName: VarAccess[39] Predecessor: [VarAccess[34]]
NodeName: VarAccess[34] Predecessor: [TypeAccess[33]]
NodeName: TypeAccess[33] Predecessor: [Entry[13]]
NodeName: Entry[13] Predecessor: []
